src/Symfony/Component/Scheduler/RecurringMessage.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Scheduler;
+
+use Symfony\Component\Scheduler\Exception\LogicException;
+use Symfony\Component\Scheduler\Trigger\CallbackTrigger;
+use Symfony\Component\Scheduler\Trigger\CronExpressionTrigger;
+use Symfony\Component\Scheduler\Trigger\PeriodicalTrigger;
+use Symfony\Component\Scheduler\Trigger\TriggerInterface;
+
+/**
+ * @experimental
+ */
+final class RecurringMessage
+{
+    private function __construct(
+        private readonly TriggerInterface $trigger,
+        private readonly object $message,
+    ) {
+    }
+
+    /**
+     * Uses a relative date format to define the frequency.
+     *
+     * @see https://www.php.net/manual/en/datetime.formats.relative.php
+     */
+    public static function every(string $frequency, object $message, \DateTimeImmutable $from = new \DateTimeImmutable(), ?\DateTimeImmutable $until = new \DateTimeImmutable('3000-01-01')): self
```

Comment:
static instead of self on all method (matters when making it non-final)

Reply:
oc